### PR TITLE
[feature] Allow configuration of alert key

### LINKED ISSUE #37: [Feature Request] Make ALERT_KEY configurable
I'm happy to take a stab at this, but would like to collaborate on how to implement before doing so. Here's what I'm thinking:
1) Add an alert_key like to the config.yml file passed to interferon, like so:
```
destinations:
  - type: datadog
    enabled: true
    options:
      alert_key: "My custom alert key"
```
2) Define a new instance method `Interferon::Destinations::Datadog#alert_key` that prefers the value from the config, but falls back to a constant default.

3) In lib/interferon/destinations/datadog.rb, turn `Interferon::Destinations::Datadog.generate_message` into an instance method, so it can access the `alert_key` instance method. Same for self.same_alerts, turn it into an instance method. As near as I can tell  `same_alerts` and `generate_message` are only ever called by instance methods anyway?

Thoughts?

### PR DESCRIPTION
**Pull request by mcwqy9**
Closes #37

### CODE REVIEW

**Comment by mcwqy9** [comment]
@jimmyngo I would love your feedback on this.

**Comment by jimmyngo** [comment]
I think it looks good, what's the use case here?

I was trying to find instances where we needed the same_alert method to be a class method, but I couldn't find anything.

**Comment by mcwqy9** [comment]
We use a custom alert_key to customize the branding of interferon-generated alerts in our datadog instance. In addition We have markdown in our custom alert_key that links back to internal documentation about interferon, related tools, where to get help & how to make changes, etc.

File: spec/lib/interferon_spec.rb [hunk 1]
```diff
@@ -7,6 +7,11 @@
...
   let(:the_existing_alerts) { mock_existing_alerts }
   let(:dest) { MockDest.new(the_existing_alerts) }
+  let(:datadog) do
+    Interferon::Destinations::Datadog.new(
+      'app_key' => 'TEST_APP_KEY', 'api_key' => 'TEST_API_KEY'
+    )
+  end
 
   shared_examples_for 'alert_option' do |alert_option, same_value, different_value, alert_dsl|
     let(:json_message) { 'message' + "\n#{Interferon::Destinations::Datadog::ALERT_KEY}" }
```
Comment (jimmyngo) [inline spec/lib/interferon_spec.rb]: Should replace `Interferon::Destinations::Datadog::ALERT_KEY` with `datadog.alert_key` in the rest of the file.

File: spec/lib/interferon/destinations/datadog_spec.rb [hunk 2]
```diff
@@ -148,28 +153,34 @@
...
     it 'adds the ALERT_KEY to the message' do
-      expect(Interferon::Destinations::Datadog.generate_message(message, people)).to include(
+      expect(datadog.generate_message(message, people)).to include(
         Interferon::Destinations::Datadog::ALERT_KEY
       )
     end
 
+    it 'prefers a custom alert_key if provided' do
+      expect(datadog_alert_key.generate_message(message, people)).to include(
+        'My custom alert key'
```
Comment (jimmyngo) [inline spec/lib/interferon/destinations/datadog_spec.rb]: Make this alert key a variable.

**Pull request merged**